sdk/core/core-rest-pipeline/src/policies/auxiliaryAuthenticationPolicy.ts

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { GetTokenOptions, TokenCredential } from "@azure/core-auth";
+import { AzureLogger } from "@azure/logger";
+import { PipelineRequest, PipelineResponse, SendRequest } from "../interfaces";
+import { PipelinePolicy } from "../pipeline";
+import { createTokenCycler } from "../util/tokenCycler";
+import { logger as coreLogger } from "../log";
+import { AuthorizeRequestOptions } from "./bearerTokenAuthenticationPolicy";
+
+/**
+ * The programmatic identifier of the bearerTokenAuthenticationPolicy.
+ */
+export const auxiliaryAuthenticationPolicyName = "auxiliaryAuthenticationPolicy";
+const AUTHORIZATION_AUXILIARY_HEADER = "x-ms-authorization-auxiliary";
+
+/**
+ * Options to configure the auxiliaryAuthenticationPolicy
+ */
+export interface AuxiliaryAuthenticationPolicyNameOptions {
+  /**
+   * The TokenCredential list that can supply the auxiliary authentication token.
+   */
+  credentials?: TokenCredential[];
```

Comment:
Even there is a limitation expressed per the docs but I didn't notice any conditions checks for this cross other languages' implementation.

I would prefer not to add any limitations either. Because the detailed scenarios would require # of tokens e.g creating a peer between virtual network would require one extra token, passing more than one credentials here would have no benifit in customer code. So customer could be responsible for this.

Reply:
see comments: https://github.com/Azure/azure-sdk-for-js/pull/25270#discussion_r1176716752